ArduCopter/mode_auto.cpp

```diff
@@ -335,19 +335,18 @@ void ModeAuto::takeoff_start(const Location& dest_loc)
         alt_target_terrain = true;
     } else {
         // set horizontal target
-        Location dest(dest_loc);
-        dest.lat = copter.current_loc.lat;
-        dest.lng = copter.current_loc.lng;
+        takeoff_loc = Location(copter.current_loc.lat, copter.current_loc.lng, dest_loc.alt, dest_loc.get_alt_frame());
```

Comment:
Doesn't this leave the above-terrain case with the same problem you're fixing here?

Reply:
probably if you move the drone enough between the mission loading and the arming. Otherwise, as it it relative to the terrain, it shouldn't be an issue. We would need autotest for those case.